activerecord/test/cases/associations/inverse_associations_test.rb

```diff
@@ -802,6 +802,44 @@ def test_with_hash_many_inversing_does_not_add_duplicate_associated_objects
     end
   end
 
+  def test_with_has_many_inversing_does_not_add_unsaved_duplicate_records_when_collection_is_unloaded
```

Comment:
Since the bug was to specific to duplicate records in the case where the collection was loaded via `#load`, do we need to add tests for the case where the collection is unloaded? (ie. this test and the one on L824). Seems like they pass without the changes in this PR, and are potentially redundant with some of the other tests in this file. What do you think?

Reply:
It was added to investigate a problem, but I removed it because it is unnecessary. 👍